### DISCUSSION TITLE
Remix deploys throw Error: ENOENT: no such file or directory, scandir 'app/routes'

### DISCUSSION THREAD

**Discussion by agcty** [vercel/vercel#9478] [General]
All my Remix deploys are failing with the above error. Last working Vercel CLI version was 28.15.3, since then, all deploys fail. Can't even redeploy a known working commit anymore.

**Comment by WesleyYue** [comment]
Also see this issue, but I'm setting up a remix app in a turborepo for the first time. I tried deploying this [example](https://github.com/remix-run/examples/tree/main/turborepo-vercel) and it seems to work fine. I wonder if having remix-flat-routes in my project is breaking this somehow... debugging the differences between my repo and the example repo right now.

**Reply by WesleyYue** [reply]
Yep, this is definitely the issue for me. I made this change to remix.config.js from the example repo:
```

  // ignoredRouteFiles: [".*"],
  ignoredRouteFiles: ["**/*"],
  routes: async (defineRoutes) => {
    return flatRoutes("routes", defineRoutes);
  },
```

And got `Error: ENOENT: no such file or directory, scandir 'app/routes'`

**Reply by WesleyYue** [reply]
Posted about it here https://github.com/kiliman/remix-flat-routes/issues/42

**Reply by WesleyYue** [reply]
Solution from https://github.com/kiliman/remix-flat-routes/issues/42#issuecomment-1436409214

```javascript
routes: async (defineRoutes) => {
    return flatRoutes("routes", defineRoutes, {
        appDir: path.resolve(__dirname, 'app')
    });
},
```

**Comment by TooTallNate** [comment]
This was a regression from a recent change. Fix is in https://github.com/vercel/vercel/pull/9518. Will follow up here when it's released.

**Comment by agcty** [comment]
28.16.6 fixes the app/routes issue, but now there's a new one 🥲 @TooTallNate :



```
[Error: ENOENT: no such file or directory, readlink '/vercel/output/functions/.func'] {
--
08:52:57.206 | errno: -2,
08:52:57.207 | code: 'ENOENT',
08:52:57.207 | syscall: 'readlink',
08:52:57.207 | path: '/vercel/output/functions/.func'
08:52:57.207 | }
```
